libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -807,9 +807,63 @@ MAV_MISSION_RESULT AP_Mission::mavlink_to_mission_cmd(const mavlink_mission_item
     return MAV_MISSION_ACCEPTED;
 }
 
+// converts a mission_item to mission_item_int and returns a mission_command
+MAV_MISSION_RESULT AP_Mission::mavlink_to_mission_cmd(const mavlink_mission_item_t& packet, AP_Mission::Mission_Command& cmd)
+{
+    mavlink_mission_item_int_t mav_cmd = {};
+
+    mav_cmd.param1 = packet.param1;
+    mav_cmd.param2 = packet.param2;
+    mav_cmd.param3 = packet.param3;
+    mav_cmd.param4 = packet.param4;
+    mav_cmd.z = packet.z;
+    mav_cmd.seq = packet.seq;
+    mav_cmd.command = packet.command;
+    mav_cmd.target_system = packet.target_system;
+    mav_cmd.target_component = packet.target_component;
+    mav_cmd.frame = packet.frame;
+    mav_cmd.current = packet.current;
+    mav_cmd.autocontinue = packet.autocontinue;
+    
+    // convert to int
+    mav_cmd.x = packet.x * 1.0e7f;
+    mav_cmd.y = packet.y * 1.0e7f;
+    
+    MAV_MISSION_RESULT ans = mavlink_int_to_mission_cmd(mav_cmd, cmd);
+    
+    return ans;
+}
+
+// converts a Mission_Command to mission_item_int and returns a mission_item
+bool AP_Mission::mission_cmd_to_mavlink(const AP_Mission::Mission_Command& cmd, mavlink_mission_item_t& packet)
+{
+    mavlink_mission_item_int_t mav_cmd = {};
+    
+    bool ans = mission_cmd_to_mavlink_int(cmd, (mavlink_mission_item_int_t&)mav_cmd);
+    
+    packet.param1 = mav_cmd.param1;
+    packet.param2 = mav_cmd.param2;
+    packet.param3 = mav_cmd.param3;
+    packet.param4 = mav_cmd.param4;
+    packet.z = mav_cmd.z;
+    packet.seq = mav_cmd.seq;
+    packet.command = mav_cmd.command;
+    packet.target_system = mav_cmd.target_system;
+    packet.target_component = mav_cmd.target_component;
+    packet.frame = mav_cmd.frame;
+    packet.current = mav_cmd.current;
+    packet.autocontinue = mav_cmd.autocontinue;
+    
+    // convert to float
+    packet.x = mav_cmd.x / 1.0e7f;
```

Comment:
Same comment as above that this is not valid for local coord frames which are 1e4 as noted above.